src/sinol_make/commands/run/__init__.py

```diff
@@ -368,48 +377,98 @@ def print_view(output_file=None):
+			programs = list(set(programs))
 		compilation_results = self.compile_programs(programs)
 		os.makedirs(self.EXECUTIONS_DIR, exist_ok=True)
 		compiled_commands = []
-		for subtask in self.config["subtasks"]:
-			validator_program = os.path.join(self.EXECUTABLES_DIR, self.config["subtasks"][subtask]["validator"])
-			compiled_commands.append((subtask, validator_program, True))
-		names = list(self.config["subtasks"])
-		results = self.perform_executions(compiled_commands, names, programs, self.args.subtask_report)
-		for subtask in self.config["subtasks"]:
-			passed_groups = []
-			for group in self.config["scores"]:
-				if results[subtask][group] == "OK":
-					passed_groups.append(group)
-			passed_groups.sort()
-			should_pass = self.config["subtasks"][subtask]["groups"]
-			should_pass.sort()
-			if passed_groups != should_pass:
-				print(util.bold(util.color_red('Subtask %s will pass groups %s (expected %s).' % (subtask, passed_groups, should_pass))))
-				exit(1)
+		if suggestions:
+			for program in programs:
+				path = os.path.join(self.EXECUTABLES_DIR, program)
+				compiled_commands.append((program, path, True))
+			names = programs
+			results = self.perform_executions(compiled_commands, names, programs, self.args.expected_scores_report)
+
+			print(util.bold("Suggested expected scores description:"))
+			print("sinol_expected_scores:")
+
+			i = 1
+			new_config = self.config
+			new_config["sinol_expected_scores"] = {}
+
+			for program in programs:
+				print("  %s:" % program)
+				print("    program: %s" % program)
+				print("    expected: {" + ', '.join(results[program].items()) + "}")
+				new_config["sinol_expected_scores"][program] = {
+					"program": program,
+					"expected": results[program],
+					"points": 0
+				}
+
+				points = 0
+				for group, result in results[program].items():
+					if result == "OK":
+						points += self.scores[group]
+				print("    points: %d" % points)
+				new_config["sinol_expected_scores"][program]["points"] = points
+
+			if self.args.apply_suggestions:
+				print("Do you want to apply these suggestions? (y/n) ", end="")
+				choice = input().lower()
+				if choice == "y" or choice == "yes" or choice == "":
+					with open(os.path.join(os.getcwd(), "config.yml"), "w") as f:
+						yaml.dump(new_config, f, default_flow_style=False)
+					print(util.color_green("Suggestions applied."))
+				else:
+					print("Suggestions not applied.")
+		else:
+			for prog in self.config["sinol_expected_scores"]:
+				program = os.path.join(self.EXECUTABLES_DIR, self.config["sinol_expected_scores"][prog]["validator"])
+				compiled_commands.append((prog, program, True))
+			names = list(self.config["sinol_expected_scores"])
+			results = self.perform_executions(compiled_commands, names, programs, self.args.expected_scores_report)
+			for prog in self.config["sinol_expected_scores"]:
+				for group, expected_result in self.config["sinol_expected_scores"][prog]["expected"].items():
```

Comment:
The `self.config["sinol_expected_scores"][prog]` shows up often and it makes the code a bit unreadable (because this long line takes most of my attention but it's just an implementation detail and is irrelevant)

Reply:
Maybe I should add const `EXPECTED_SCORES = “sinol_expected_scores”` and use it?

Reply:
That won't make the code really shorter -- the string still is there, I still have to read it every three lines. I guess that first saving it to a variable and using this variable instead of `self.config["sinol_expected_scores"][prog]` could be better.

Reply:
It should be more readable now.